tide_landing_page.module

```diff
@@ -175,4 +179,83 @@ function tide_landing_page_field_widget_paragraphs_form_alter(&$element, FormSta
       }
     }
   }
+  if ($paragraph_field_name == 'field_landing_page_component') {
+    $widget_state = WidgetBase::getWidgetState($element['#field_parents'], $paragraph_field_name, $form_state);
+    $paragraph = $widget_state['paragraphs'][$element['#delta']]['entity'];
+    $paragraph_type = $paragraph->bundle();
+    // Add states for navigation and promotion card paragraph type fields.
+    if ($paragraph_type == 'navigation_card' || $paragraph_type == 'promotion_card') {
+      $dependee_field_name = 'field_paragraph_link';
+      $selector = sprintf(':input[name^="%s[%d][subform][%s][0][uri]"]', $paragraph_field_name, $element['#delta'], $dependee_field_name);
+      // Dependent fields.
+      if (isset($element['subform']['field_paragraph_summary'])) {
+        $element['subform']['field_paragraph_summary']['widget'][0]['value']['#states'] = [
+          'visible' => [
+            $selector => [
+              ['value' => ['pattern' => '^http']],
+              ['value' => ''],
+            ],
+          ],
+        ];
+        $element['subform']['summary_message_group'] = [
+          '#type' => 'fieldset',
+          '#title' => 'Page summary will be automatically drawn from the content you have linked to.',
+          '#group_name' => 'Page summary message',
+          '#attributes' => [
+            'class' => 'summary-message-group',
+          ],
+          '#states' => [
+            'invisible' => [
+              $selector => [
+                ['value' => ['pattern' => '^http']],
+                ['value' => ''],
+              ],
+            ],
+          ],
+        ];
+      }
+      if (isset($element['subform']['field_customise'])) {
+        $element['subform']['customise_field_group'] = [
+          '#type' => 'fieldset',
+          '#title' => 'Customise',
+          '#group_name' => 'Page summary message',
+          '#attributes' => [
+            'class' => 'customise-field-group',
+          ],
+          '#weight' => '20',
+          '#states' => [
+            'invisible' => [
+              $selector => [
+                ['value' => ['pattern' => '^http']],
+                ['value' => ''],
+              ],
+            ],
+          ],
+        ];
+        $element['subform']['field_customise']['#states'] = [
+          'invisible' => [
+            $selector => [
+              ['value' => ['pattern' => '^http']],
+              ['value' => ''],
+            ],
+          ],
+        ];
+        $element['subform']['customise_field_group']['field_customise'] = $element['subform']['field_customise'];
+        unset($element['subform']['field_customise']);
```

Comment:
Hey @MdNadimHossain can you do a test for me: Can you submit the form and see if this value is saving correctly?
If it is saving correctly, all good. If not then you would need a check in the submit handler to add this value back into $element['subform']['field_customise'] as this value maybe altered from a hook but database code may still be looking at the previous field name.

Reply:
@asarfara yes looks like it saved correctly. I have just checked this option and saved the node. Then when I go to node/edit, I still can see that the field is checked.
![Screen Shot 2020-11-11 at 9 39 02 am](https://user-images.githubusercontent.com/20810541/98742756-0efe9800-2403-11eb-8fda-a0600b9640c0.png)

Reply:
Awesome :D Win Win :D